pkgs/development/compilers/zig/generic.nix

```diff
@@ -64,6 +65,9 @@ stdenv.mkDerivation (finalAttrs: {
   };
 
   meta = {
+    # Zig does not support Apple Silicon on Linux yet.
+    # Ref: https://github.com/ziglang/zig/issues/16331
+    broken = stdenv.isLinux && targetPlatform.isAppleSilicon;
```

Comment:
This is a conflation I don't think we should be making — whether Zig is broken doesn't depend on whether we're specifically building for Apple Silicon.  The package is equally broken on Apple Silicon when built with the generic CPU model.

Reply:
That is true but I don't see how we could determine the package is broken and the model is generic but being built on Apple Silicon. I mostly did it as a placeholder for writing a patch that makes Zig working on Apple Silicon and to test the `isAppleSilicon` attribute.

Reply:
But such a patch needs to be applied for all aarch64-linux — we can't force users to specify their CPU and rebuild the world just to have programs that work on their CPU.

Reply:
Can't we just always build Zig on aarch64 for a target with 64K pages, then?

Reply:
Maybe but who knows what other issues that could bring

Reply:
Yeah though this feels like it kinda gets off topic on this particular PR. Could raise an issue with the maintainers and get their thoughts.

Reply:
I agree, but I'm bringing it up because I don't think that isAppleSilicon has really been justified.  Simplifying specifying CPU-specific compiler flags seems good, but I don't think it's ever really going to be the right thing to have conditionals based on them.

Reply:
So should we remove it from this PR or just keep it as is until a follow up PR is made to actually fix Zig?

Reply:
We should remove it from this PR.  There should be no predicates for CPU type.

(Another reason I thought of overnight: Apple Silicon doesn't even necessarily imply 16K pages! You can run a 4K kernel, although lots of stuff currently won't work, it's not some fundamental limitation of the CPU.)

Reply:
Done

Reply:
No it isn't?  The CPU type predicates are still there.

Reply:
Where? I don't see it in the Zig files.

Reply:
Ah, when I say that "there should be no predicates for CPU type", I meant that they shouldn't exist, not just that they shouldn't be used in the Zig package.  As I said, I would be surprised if having conditionals based on target CPU (as opposed to passing it to a compiler, etc.) is ever the right thing to do.

Reply:
Alright, done.